Library/Formula/linenoise.rb

```diff
@@ -0,0 +1,51 @@
+class Linenoise < Formula
+  homepage "https://github.com/antirez/linenoise"
+  url "https://github.com/antirez/linenoise/archive/1.0.tar.gz"
+  sha256 "f5054a4fe120d43d85427cf58af93e56b9bb80389d507a9bec9b75531a340014"
+
+  # Added patch to build linenoise as a dynamic library
+  # Upstream Issue: https://github.com/antirez/linenoise/issues/94
+  patch :DATA
```

Comment:
This patch is OS X centric and the upstream library is not, so I don't feel that it's legit as an upstream pull-request. I don't have the expertise to build such a patch.